javascript/ql/lib/semmle/javascript/dataflow/AdditionalFlowSteps.qll

```diff
@@ -1,4 +1,259 @@
+/**
+ * This contains three step-contribution classes, in order to support graceful deprecation of the old data flow library.
+ *
+ * - `class AdditionalFlowStep`: steps used only by the new dataflow library
+ * - `class LegacyFlowStep`: steps used only by the old data flow library
+ * - `class SharedFlowStep`: steps used by both
+ *
+ * The latter two will be deprecated in the future, but are currently not marked as `deprecated`.
+ * This is because a library model should be able to support both data flow libraries simultaneously, without itself getting
+ * deprecation warnings.
```

Comment:
I'm not sure I understand. What does it mean to mark a library model as deprecated?

Reply:
`class LegacyFlowStep` is currently not marked as deprecated, because you want to have library models that extend `LegacyFlowStep` without the library model getting deprecation warnings.  

However, could you deprecate `LegacyFlowStep` if you also mark all the classes that extend `LegacyFlowStep` (in library models) as deprecated?